New post "Migrating kafka with MirrorMaker2"

rmarting opened the pull request:
Proposal for a new blog post "Migrating Kafka clusters with MirrorMaker2 and Strimzi".

Content based in my own personal [blog post](https://blog.jromanmartin.io/2021/11/19/migrating-kafka-with-mirror-maker2.html) and ready for the review of the Strimzi community.

Authors: Roman Martin (rmarting), and Manuel Schindler (maschind)

maschind commented:
I hope the strimzi community will like it. Very relevant and I used the scenario already in the field!

scholzj commented on _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:
```diff
@@ -0,0 +1,227 @@
+executes this complex process of migrating and synchronizing both
+Kafka clusters.
+
+## Red Hat AMQ Streams Operators make migrations happier
```
Maybe we can talk about Strimzi here instead of `Red Hat AMQ Streams`.

rmarting replied:
Sure! It was the original approach, but ... I forgot these references. :thinking:

scholzj commented on _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:
```diff
@@ -0,0 +1,227 @@
+[Kubernetes Operators](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+provided by the great community of [Strimzi](https://strimzi.io/).
+
+Red Hat AMQ Streams provides a full list of operators to help streamline
```
Again, using Strimzi would be probably the proper way here.

rmarting replied:
Idem as above! Fixing!

scholzj reviewed (commented):
Thanks for the PR. This looks real useful. I left few comments. We will also (on the Strimzi side) need to look into how we can fix the multiple authors issue.

rmarting commented on _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:
```diff
@@ -0,0 +1,227 @@
+layout: post
+title: "Migrating Kafka clusters with MirrorMaker2 and Strimzi"
+date: 2021-11-09
+author: roman_martin,manuel_schindler
```
Will it be a blocker to publish the blog? If so, we could remove one of the authors (e.g. myself) when you accepted this blog.

scholzj replied:
I'm sure it is not a blocker from our side. It is more that we want to give you the credit ;-)

If you would be fine with it, maybe the easiest way would be that instead of having two authors to have just one author with two names. Something like this:
```yaml
roman_martin_manuel_schindler:
  name: Roman Martin, Manuel Schindler
```

That would get both names there, just no link.

rmarting replied:
I liked your approach! Fixed in that way!

scholzj reviewed (approved):
Nice post. Thanks.

rmarting commented:
Thank you so much to accept us our small contribution to the Strimzi Blog!!!

scholzj commented:
Just FYI: It still needs the approval of others.

ppatierno commented on _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:
```diff
@@ -0,0 +1,227 @@
+* **MirrorMaker 1**: The first implementation with basic functionality
+to consume and produce data from the source to the target. It is very
+limited and it could help you to replicate only data payloads. 
+* **MirrorMaker 2**: Improved implementation based in
```
"based on" ?

ppatierno commented on _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:
```diff
@@ -0,0 +1,227 @@
+from the target Kafka cluster at the same point the consumers stopped in
+the source Kafka cluster. This feature tries to avoid consuming duplicated
+messages in the target cluster.
+* **Dynamic detection of changes** in source Kafka cluster and replicate in
```
we are talking about detection of configuration changes on brokers right? should be clearer here?

ppatierno reviewed (approved):
First of all thanks for the contribution! very nice article!
I am approving it but I left a couple of nits and I have another point to raise: I think that the article could talk about Kubernetes in general, because it's going to work on it, not needed to focus on OpenShift as it was a pre-requisite.
Maybe it should refer to Kubernetes but then explain that your specific use case was on OpenShift which is an enterprise distribution of Kubernetes by Red Hat. 
At least this is my view, maybe other maintainers could have a different one.

scholzj commented:
> I am approving it but I left a couple of nits and I have another point to raise: I think that the article could talk about Kubernetes in general, because it's going to work on it, not needed to focus on OpenShift as it was a pre-requisite.
Maybe it should refer to Kubernetes but then explain that your specific use case was on OpenShift which is an enterprise distribution of Kubernetes by Red Hat.
> At least this is my view, maybe other maintainers could have a different one.

I don't know. I was thinking about that part as well. The AMQ Streams references didn't seemed suitable on a Strimzi blog since Strimzi should be the focus. But OpenShift is one of the platforms we support and which many users use. So that seems fine to me. That said, I think it might be worth mentioning at the end or in the beginning that this should work on other Kube distros as well.

ppatierno commented:
I didn't read about "AMQ Streams" on OpenShift but more "Strimzi" on OpenShift but now I noticed that the repo with the demo is named as AMQ Streams even if it refers to Strimzi AFAIU.

scholzj commented:
>I didn't read about "AMQ Streams" on OpenShift but more "Strimzi" on OpenShift but now I noticed that the repo with the demo is named as AMQ Streams even if it refers to Strimzi AFAIU.

It is just a repo name, nothing else.

ppatierno commented:
> It is just a repo name, nothing else.

Yeah exactly it's totally fine with me even because anyway it talks about Strimzi in the end.
I would just change the article here to talk more about Kubernetes or as you said making clear that the use case was on OpenShift but everything would work with Kubernetes. Anyway the things to change from examples would be the external listeners not using "route".

rmarting commented:
Thanks @ppatierno and @scholzj for your feedback and comments.

I agree that this article should be aligned with the upstream community of Strimzi, so:

- Removed any reference about Red Hat AMQ Streams
- Added a comment to clarify that the article is based in OpenShift, but it works in Kubernetes too
- Renamed the sample repo to strimzi-migration-demo (main branch for Strimzi and a rh-amq-streams branch for the product)

I hope these changes helped for your review, and approvement.

ppatierno commented:
@rmarting The changes look good to me. If @scholzj agrees we could publish this article tomorrow, but it means you have to change the date (which currently at 2021-11-09).

scholzj commented:
The date can be fixed while merging it, no need to worry about it. Tomorrow sounds good.

scholzj commented:
Thanks for the blog post @rmarting & @maschind

rmarting commented:
Thanks Strimzi Community to accept our small contribution!!!
